src/m365/entra/commands/group/group-set.ts

```diff
@@ -0,0 +1,292 @@
+            return `User principal name '${isValidUserPrincipalNameArray}' is invalid for option 'memberUserNames'.`;
+          }
+        }
+
+        if (args.options.visibility && ['Public', 'Private'].indexOf(args.options.visibility!) === -1) {
+          return `Option 'visibility' must be one of the following values: Public, Private.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'displayName'] }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let groupId = args.options.id;
+    let ownerIds: string[] = [];
+    let memberIds: string[] = [];
+
+    try {
+      if (args.options.displayName) {
+        if (this.verbose) {
+          await logger.logToStderr(`Retrieving group id...`);
+        }
+
+        groupId = await entraGroup.getGroupIdByDisplayName(args.options.displayName);
+      }
+
+      const manifest = this.createRequestBody(args.options);
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/groups/${groupId}`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        data: manifest
+      };
+
+      ownerIds = await this.getUserIds(logger, args.options.ownerIds, args.options.ownerUserNames);
+      memberIds = await this.getUserIds(logger, args.options.memberIds, args.options.memberUserNames);
+
+      await request.patch(requestOptions);
+
+      if (ownerIds.length !== 0) {
+        await this.addUsers(groupId!, 'owners', ownerIds);
+      }
+
+      if (memberIds.length !== 0) {
+        await this.addUsers(groupId!, 'members', memberIds);
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  };
+
+  private createRequestBody(options: Options): any {
+    const requestBody: any = {
+      displayName: options.newDisplayName,
+      description: options.description,
+      mailNickName: options.mailNickname,
+      visibility: options.visibility
+    };
+
+    this.addUnknownOptionsToPayload(requestBody, options);
+    return requestBody;
+  }
+
+  private async getUserIds(logger: Logger, userIds: string | undefined, userNames: string | undefined): Promise<string[]> {
+    if (userIds) {
+      return userIds.split(',').map(o => o.trim());
+    }
+
+    if (!userNames) {
+      if (this.verbose) {
+        await logger.logToStderr('No users to validate, skipping.');
+      }
+      return [];
+    }
+
+    if (this.verbose) {
+      await logger.logToStderr('Retrieving user information.');
+    }
+
+    const userArr: string[] = userNames.split(',').map(o => o.trim());
+
+    if (this.verbose) {
+      await logger.logToStderr('Retrieving ID(s) of user(s)...');
+    }
+
+    return entraUser.getUserIdsByUpns(userArr);
+  }
+
+  private async addUsers(groupId: string, role: string, userIds: string[]): Promise<void> {
```

Comment:
Right now we are only adding users. Since this is a `set` command, the aim is that the specified users are becoming the only users.
This means:
- Users that are not specified are removed as owner or member
- Users that are already part of the group stay owner or member
- Users that were not part of the group, will become a new owner or member

So wrapping up, if I specify 'Kevin, Megan and Joe' as member. I expect that these 3 people are the only members of the group. All other people are removed. Same goes for the owners.
If only members are specified, we modify the members and leave the owners untouched and vice versa.
Hopefully this clairifies the intention of this command a bit.

Reply:
Thanks for clarification.